kratos/sources/model_part_io.cpp

```diff
@@ -1774,6 +1775,60 @@ namespace Kratos
 
         KRATOS_CATCH("")
     }
+    template<class TDataType>
+    void ModelPartIO::WriteDataBlock(const TDataType& rThisData, const std::string& DataName){
+        typedef VariableComponent<VectorComponentAdaptor<array_1d<double, 3> > > array_1d_component_type;
+        std::unordered_set<std::string> variables;
+        
+        for(auto& data :rThisData){
```

Comment:
It would be better to use `object` and `TObjectsContainerType` instead of `data` and `TDataType`

Reply:
okay, done